src/hooks/useChatStream.ts

```diff
@@ -71,6 +64,43 @@ const useChatStream = (input: UseChatStreamInput) => {
     return { ...initialMessage, content: response };
   };
 
+  const fetchAndUpdateAIResponseWithMetadata = async (message: string) => {
+    const charactersPerSecond = input.options.fakeCharactersPerSecond;
+    const stream = await getStream(message, input.options, input.method);
+    const initialMessage = addMessage({ content: '', role: 'bot' });
+    let response = '';
+    let metadata = null;
+
+    for await (const chunk of decodeStreamToJson(stream)) {
+      const jsonObjects = extractJsons(chunk);
+
+      for (const parsedChunk of jsonObjects) {
+        if (parsedChunk.type === 'content') {
```

Comment:
Yeah lets document this expected implementation detail like we discussed in Slack 👍🏻